Sources/PackageCollections/PackageIndexAndCollections.swift

```diff
@@ -0,0 +1,284 @@
+    ) {
+        self.collections.getCollection(source, callback: callback)
+    }
+
+    public func listPackages(
+        collections: Set<PackageCollectionsModel.CollectionIdentifier>?,
+        callback: @escaping (Result<PackageCollectionsModel.PackageSearchResult, Error>) -> Void
+    ) {
+        self.collections.listPackages(collections: collections, callback: callback)
+    }
+
+    public func listTargets(
+        collections: Set<PackageCollectionsModel.CollectionIdentifier>?,
+        callback: @escaping (Result<PackageCollectionsModel.TargetListResult, Error>) -> Void
+    ) {
+        self.collections.listTargets(collections: collections, callback: callback)
+    }
+    
+    public func findTargets(
+        _ query: String,
+        searchType: PackageCollectionsModel.TargetSearchType?,
+        collections: Set<PackageCollectionsModel.CollectionIdentifier>?,
+        callback: @escaping (Result<PackageCollectionsModel.TargetSearchResult, Error>) -> Void
+    ) {
+        self.collections.findTargets(query, searchType: searchType, collections: collections, callback: callback)
+    }
+    
+    // MARK: - Package index APIs
+
+    public func isIndexEnabled(callback: @escaping (Result<Bool, Error>) -> Void) {
+        self.index.get { result in
+            switch result {
+            case .failure(PackageIndexError.featureDisabled):
+                callback(.success(false))
+            case .failure(let error):
+                callback(.failure(error))
+            case .success(let url):
+                callback(.success(url != .none))
+            }
+        }
+    }
+
+    public func setIndex(
+        url: Foundation.URL,
+        callback: @escaping (Result<Void, Error>) -> Void
+    ) {
+        self.index.set(url: url, callback: callback)
+    }
+
+    public func unsetIndex(callback: @escaping (Result<Void, Error>) -> Void) {
+        self.index.unset(callback: callback)
+    }
+    
+    public func getIndex(callback: @escaping (Result<Foundation.URL?, Error>) -> Void) {
+        self.index.get(callback: callback)
+    }
+    
+    public func listPackagesInIndex(
+        offset: Int,
+        limit: Int,
+        callback: @escaping (Result<PackageCollectionsModel.PaginatedPackageList, Error>) -> Void
+    ) {
+        self.index.listPackages(offset: offset, limit: limit, callback: callback)
+    }
+    
+    // MARK: - APIs that make use of package index and collections
+    
+    public func getPackageMetadata(
+        identity: PackageIdentity,
+        location: String?,
+        collections: Set<PackageCollectionsModel.CollectionIdentifier>?,
+        callback: @escaping (Result<PackageCollectionsModel.PackageMetadata, Error>) -> Void
+    ) {
+        // Get metadata using both package index and collections
+        let sync = DispatchGroup()
+        let results = ThreadSafeKeyValueStore<Source, Result<PackageCollectionsModel.PackageMetadata, Error>>()
+        DispatchQueue.sharedConcurrent.async {
```

Comment:
since getPackageMetadata is already async, invoking it in  the DispatchQueue.sharedConcurrent.async block seems redundant?